deploy/bolt_db_store.go

```diff
@@ -38,201 +38,156 @@ func NewBoltDBStore(path string) (*BoltDBStore, error) {
 	return &BoltDBStore{db: db}, nil
 }
 
-func (s *BoltDBStore) Get(key string) (deploy Deploy, ok bool) {
+func (s *BoltDBStore) GetQueue(key string) (queue Queue) {
+	ok := false
+
 	s.db.View(func(tx *bolt.Tx) error {
-		b := tx.Bucket([]byte(key))
-		if b == nil {
-			return nil
-		}
+		bucket := tx.Bucket([]byte(key))
 
-		lastDeployKey, _ := b.Cursor().Last()
-		if lastDeployKey == nil {
-			ok = false
+		if bucket == nil {
 			return nil
 		}
 
-		var err error
-		if deploy, err = s.readDeploy(lastDeployKey, b); err != nil {
-			if err == ErrNoDeploy {
-				ok = false
-				return nil
-			}
+		bytes := bucket.Get([]byte("queue"))
 
-			return err
+		if bytes == nil {
+			return nil
 		}
 
-		ok = true
+		err := json.Unmarshal(bytes, &queue)
+
+		ok = err == nil
```

Comment:
It sets a flag that is being checked later on in order to return an empty queue if the one stored in the db is corrupted or something.

Reply:
ok I got this, but still in case of an error the function will return nil. Errors should be handled or returned. Here none of this 2 happens. Do i miss something?

Reply:
I don't think it really matters that much, because this code is inside a closure and returning `nil` instead of an error just tells the database to commit a transaction instead of rolling it back. For some reason, this approach was used in the original code, so I just went with it for the sake of consistency.